ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Input.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ *
+ */
+
+package org.opensearch.ml.engine.algorithms.anomalylocalization;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.search.aggregations.AggregationBuilder;
+
+import lombok.Data;
+
+/**
+ * Information about aggregate, time, etc to localize.
+ */
+@Data
+public class Input {
+
+    private final String indexName; // name of the data index
```

Comment:
Do we plan to support index pattern like `abc*` ?

Reply:
From`AnomalyLocalizerImpl`, index name used in search request directly, so index pattern is ok.